src/Form/Form.php

```diff
@@ -471,6 +471,11 @@ protected function updateQuestions(): void
                 $uuid = null;
             }
 
+            // Extra data must be JSON encoded
+            if (isset($question_data['extra_data'])) {
+                $question_data['extra_data'] = json_encode($question_data['extra_data']);
+            }
+
```

Comment:
Do we need a safety here ?

Since this is a "free" json field, maybe each question type should have a "validateExtraDataInput()" method that make sure the correct values are supplied and throw an exception otherwise.

This would avoid setting invalid data and help catch bugs quickly.